net/softethervpn/Makefile

```diff
@@ -0,0 +1,131 @@
+endef
+
+define Build/Compile
+	# Softether compiles first a tool called hamcorebuilder which is essentially an archiver. Then the compilation calls this archiver to pack all 
+	# resource files (*.html, *.config, .txt, readme, languages) it needs to run in an archive file called hamcore.se2. The three executables are then built 
+	# and they will have a runtime dependency on this resource archive. By default OpenWRT build process cross compiles all targets in the makefile
+	# generating therefore an error when the archive is generated since the executable is crosscompiled to run on the router and instead it is run
+	# by the build host. The workaround is to compile the archiver with the host environment, generate the hamcore.se2 archive and cross compile everything
+	# again without regenerating the archive.
+
+	# we clean the last cross compiled version
+	make -C $(PKG_BUILD_DIR) clean
+
+	# we compile the hamcorebuilder utility and generate hamcore archive using host environment
+	make -C $(PKG_BUILD_DIR) src/bin/BuiltHamcoreFiles/unix/hamcore.se2
+
+	# we clean everything, but we save the built hamcore archive and we update its timestamp to avoid rebuilding the archive with cross compiled hamcorebuilder
+	mv $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2 $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2.1
+	make -C $(PKG_BUILD_DIR) clean
+		mv $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2.1 $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2
+	touch -d "`date -d 1day`" $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix/hamcore.se2
+
+	ls -la $(PKG_BUILD_DIR)/src/bin/BuiltHamcoreFiles/unix
+
+	# we cross compile everything from the scratch
+	$(MAKE) \
+		$(TARGET_CONFIGURE_OPTS) \
+		CCFLAGS="$(SE_FLAGS) -I$(STAGING_DIR)/usr/include $(ICONV_CFLAGS)" \
+		LDFLAGS="-L$(STAGING_DIR)/usr/lib $(ICONV_LDFLAGS)" \
+		-C $(PKG_BUILD_DIR)
+endef
+
+define Package/softethervpn/install
+	$(INSTALL_DIR) $(1)/usr/libexec/softethervpn
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/bin/vpnserver/vpnserver $(1)/usr/libexec/softethervpn
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/bin/vpnserver/hamcore.se2 $(1)/usr/libexec/softethervpn
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/bin/vpnclient/vpnclient $(1)/usr/libexec/softethervpn
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/bin/vpnbridge/vpnbridge $(1)/usr/libexec/softethervpn
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/bin/vpncmd/vpncmd $(1)/usr/libexec/softethervpn
+	$(INSTALL_DATA) files/dummy $(1)/usr/libexec/softethervpn/vpn_server.config
+	$(INSTALL_DATA) files/dummy $(1)/usr/libexec/softethervpn/vpn_bridge.config
+	$(INSTALL_DATA) files/dummy $(1)/usr/libexec/softethervpn/vpn_client.config
+	$(INSTALL_DATA) files/dummy $(1)/usr/libexec/softethervpn/lang.config
+
+	$(INSTALL_DIR) $(1)/usr/bin
+
+	# we can't use links because each command tries to load hamcore.se2 from the directory of the linked command (/usr/bin) and not from
+	# /usr/libexec/softether hence they all fail to start
```

Comment:
Create an `/usr/libexec/softethervpn/launcher.sh` with mode `0755` and the following contents:

    #!/bin/sh
    exec "/usr/libexec/softethervpn/${0##*/}" "$@"

Then symlink that to /usr/bin/vpn{server,bridge,client,cmd}.